packages/syft/src/syft/oblv/model.py

```diff
@@ -0,0 +1,441 @@
+        self.__logs = log_file_name
+        self.__process = process
+        return
+
+    def get_uploaded_datasets(self):
+        if len(self.client) == 0:
+            raise Exception(
+                "No Domain Clients added. Set the propert *client* with the list of your domain logins"
+            )
+        if self.__conn_string is None:
+            raise Exception(
+                "proxy not running. Use the method connect_oblv_proxy to start the proxy."
+            )
+        elif self.__conn_string == "":
+            raise Exception(
+                "Either proxy not running or not initiated using syft. Run the method initiate_connection"
+                + "to initiate the proxy connection"
+            )
+        depl = self.oblv_client.deployment_info(self.deployment_id)
+        if depl.is_deleted:
+            raise Exception(
+                "User cannot connect to this deployment, as it is no longer available."
+            )
+        req = requests.get(self.__conn_string + "/tensor/dataset/list")
+        if req.status_code == 401:
+            raise OblvUnAuthorizedError()
+        elif req.status_code == 400:
+            raise OblvEnclaveError(req.json()["detail"])
+        elif req.status_code == 422:
+            print(req.text)
+            # ToDo - Update here
+            return "Failed"
+        elif req.status_code != 200:
+            raise OblvEnclaveError(
+                "Request to publish dataset failed with status {}".format(
+                    req.status_code
+                )
+            )
+        return req.json()  # This is the publish_request_id
+
+    def publish_action(self, action: str, arguments, *args, **kwargs):
+        if self.__conn_string is None:
+            raise Exception(
+                "proxy not running. Use the method connect_oblv_proxy to start the proxy."
+            )
+        elif self.__conn_string == "":
+            raise Exception(
+                "Either proxy not running or not initiated using syft. Run the method initiate_connection"
+                + "to initiate the proxy connection"
+            )
+        depl = self.oblv_client.deployment_info(self.deployment_id)
+        if depl.is_deleted:
+            raise Exception(
+                "User cannot connect to this deployment, as it is no longer available."
+            )
+        file = None
+        if len(arguments) == 2 and arguments[1]["type"] == "tensor":
+            file = arguments[1]["value"]
+            arguments[1]["value"] = "file"
+        body = {
+            "inputs": json.dumps(arguments),
+            "args": json.dumps(args),
+            "kwargs": json.dumps(kwargs),
+        }
+        req = requests.post(
+            self.__conn_string + "/tensor/action?op={}".format(action),
+            data=body,
+            files={"file": file},
+        )
+        # req = requests.post(self.__conn_string + "/tensor/action?op={}".format(action)
```

Comment:
Q: Shouldn't those be removed?

Reply:
Solved in another local mode PR by [7536c37](https://github.com/OpenMined/PySyft/commit/7536c377339ddfb39c9a17253b7998a42def1dcc)